tests/test_content_client.py

```diff
@@ -0,0 +1,175 @@
+    return katello_client.system_info.release.split('.')[0]
+
+
+def wait_for_task(task, api):
+    duration = 60
+    poll = 15
+
+    while task['state'] not in ['paused', 'stopped']:
+        duration -= poll
+        if duration <= 0:
+            break
+        time.sleep(poll)
+
+        task = api.resource('foreman_tasks').call('show', {'id': task['id']})
+
+
+def foreman_host(client, api):
+    hostname = client.check_output('hostname -s')
+    return api.resource('hosts').call('show', {'id': hostname})
+
+
+def test_register_with_subscription_manager(katello_client, entities, api, user):
+    katello_client.run_expect([0], f"subscription-manager register --force --org={entities['organization_label']} --username={user.username} --password={user.password} --env=Library")
+    assert foreman_host(katello_client, api)
+
+
+def test_enable_content_view_repo(katello_client, entities):
+    enable = katello_client.check_output(f"subscription-manager repos --enable={entities['organization_label']}_{entities['product_label']}_{entities['yum_repository_label']}")
+    assert "is enabled for this system" in enable
+
+
+def test_install_katello_host_tools(katello_client):
+    katello_client.run_expect([0], 'yum -y install katello-host-tools')
+    katello_host_tools = katello_client.package('katello-host-tools')
+    assert katello_host_tools.is_installed
+
+
+def test_install_package_local(katello_client):
+    katello_client.run('yum -y install walrus-0.71')
+    walrus = katello_client.package('walrus')
+
+    assert walrus.is_installed
+    assert walrus.version == '0.71'
+
+
+def test_available_errata(katello_client, entities, api, os_release):
+    if os_release == "6":
+        katello_client.run_expect([0], 'yum -y install katello-host-tools')
+
+    katello_client.run('yum -y install walrus-0.71')
+    time.sleep(10)
+
+    host = foreman_host(katello_client, api)
+    errata = api.resource('errata').call('index', {'host_id': host['id']})
+    errata_ids = [erratum['errata_id'] for erratum in errata['results']]
+
+    assert entities['errata_id'] in errata_ids
+
+
+@pytest.mark.katello_agent
+def test_install_katello_agent(katello_client, os_release):
+    if os_release == "8":
+        pytest.skip("katello-agent not available on EL8")
+
+    katello_client.run_expect([0], 'yum -y install katello-agent')
+    katello_agent = katello_client.package('katello-agent')
+
+    assert katello_agent.is_installed
+
+
+@pytest.mark.katello_agent
+def test_katello_agent_package_install(katello_client, api, os_release):
+    if os_release == "8":
+        pytest.skip("katello-agent not available on EL8")
+
+    katello_client.run_expect([0], 'yum -y install katello-agent')
+    host = foreman_host(katello_client, api)
+    task = api.resource('host_packages').call('install', {'host_id': host['id'], 'packages': ['gorilla']})
+    wait_for_task(task, api)
+
+    gorilla = katello_client.package('gorilla')
+    assert gorilla.is_installed
```

Comment:
I don't think `test_install_katello_agent` -- it only tests that k-agent can be installed, but we already ensured the client can install packages locally. How about converting this to a fixture, and then re-using it for the other k-agent tests, like this:

```suggestion
@pytest.fixture
def katello_agent_client(katello_client, os_release):
    if os_release not in ("6", "7"):
        pytest.skip(f"katello-agent not available on EL{os_release}")

    katello_client.run_expect([0], 'yum -y install katello-agent')
    
    katello_agent = katello_client.package('katello-agent')
    assert katello_agent.is_installed
    
    return katello_client
    

@pytest.mark.katello_agent
def test_katello_agent_package_install(katello_agent_client, api):
    host = foreman_host(katello_client, api)
    task = api.resource('host_packages').call('install', {'host_id': host['id'], 'packages': ['gorilla']})
    wait_for_task(task, api)

    gorilla = katello_client.package('gorilla')
    assert gorilla.is_installed
```